alphagov/govwifi-logging-api
# Feat/active users metrics

# Description
Add a class to generate active users metrics and upload the result to S3. It delegates the hard work to the code in `PerformancePlatform::Gateway::ActiveUsers` except without the severe indirection of that presenter/gateway/usecase pattern.

This class should be used in a Rake task like this:

```ruby
metrics = Metrics::ActiveUsers.new(period: "week", date: Date.today.to_s)

metrics.generate!

metrics.publish! 
```

The two-step invocation allows the caller to investigate metrics generation and publishing separately, should either of these need to be debugged/altered.

This is the test suite:
```
Metrics::ActiveUsers
  stores the date
  stores the period
  rejects invalid periods
  generate!
    delegates to the performance platform gateway
    stores the result in an instance variable
  key
    returns a key for the S3 upload
  publish!
    when the report is nil
      throws an error
    when the report has been generated
      writes the report into the S3 metrics bucket
```

# How to test
- read commit log;
- checkout locally;
- run `docker-compose exec app bundle exec rake publish_weekly_metrics` and observe output.
